documentation/src/main/asciidoc/search-dsl.asciidoc

```diff
@@ -85,149 +120,1023 @@ include::todo-placeholder.asciidoc[]
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/searchdsl/predicate/PredicateDslIT.java[tags=bool-minimumShouldMatchNumber]
+----
+<1> At least two "should" clauses must match for this boolean predicate to match.
+<2> The hits *should* have a `description` field matching the text `robot`.
+<3> The hits *should* have a `description` field matching the text `investigate`.
+<4> The hits *should* have a `description` field matching the text `crime`.
+<5> All returned hits will match at least two of the `should` clauses:
+their description will match either `robot` and `investigate`,
+`robot` and `crime`, `investigate` and `crime`, or all three of these terms.
+====
 
-.Using advanced projection types
+.Easily adding clauses dynamically with the lambda syntax
 ====
-[source, JAVA, indent=0]
+[source, JAVA, indent=0, subs="+callouts"]
 ----
-include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/directfieldmapping/HibernateOrmSimpleMappingIT.java[tags=projection-advanced-bean]
+include::{sourcedir}/org/hibernate/search/documentation/searchdsl/predicate/PredicateDslIT.java[tags=bool-dynamicParameters]
 ----
-[source, JAVA, indent=0]
+<1> Get a custom object holding the search parameters provided by the user through a web form, for example.
+<2> Call `.bool(Consumer)`.
+The consumer, implemented by a lambda expression, will receive a builder as an argument
+and will add clauses to that builder as necessary.
+<3> By default, a boolean predicate will match nothing if there is no clause.
+To match every document when there is no clause, add a `must` clause that matches everything.
+<4> Inside the lambda, the code is free to check conditions before adding clauses.
+In this case, we only add clauses if the relevant parameter was filled in by the user.
+<5> The hits will match the clauses added by the lambda expression.
+====
+
+[[search-dsl-predicate-simple-query-string]]
+=== `simpleQueryString`: match a user-provided query
+// Search 5 anchors backward compatibility
+[[_simple_query_string_queries]]
+
+.Matching a simple query string: AND/OR operators
+====
+[source, JAVA, indent=0, subs="+callouts"]
 ----
-include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/directfieldmapping/HibernateOrmSimpleMappingIT.java[tags=projection-advanced]
+include::{sourcedir}/org/hibernate/search/documentation/searchdsl/predicate/PredicateDslIT.java[tags=simpleQueryString-boolean]
 ----
 ====
 
-The projection DSL offers more projection types, and multiple options for each type of projection.
-To learn more about the field projection, and all the other types of projection,
-refer to the following sections.
+.Matching a simple query string: NOT operator
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/searchdsl/predicate/PredicateDslIT.java[tags=simpleQueryString-not]
```

Comment:
Do you need the `+` here?

Reply:
If I don't use `withAndAsDefaultOperator`, yes. And I'd rather have one example for each feature.